src/Meditation.AttachProcessService/Services/ProcessSnapshotCreator.cs

```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using Meditation.AttachProcessService.Models;
+using Microsoft.Diagnostics.Runtime;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Meditation.AttachProcessService.Services
+{
+    internal class ProcessSnapshotCreator : IProcessSnapshotCreator
+    {
+        public Task<IProcessSnapshot> CreateProcessSnapshotAsync(ProcessId processId, CancellationToken ct)
+        {
+            return Task.Run<IProcessSnapshot>(() =>
+            {
+                var dataTarget = DataTarget.CreateSnapshotAndAttach(processId.Value);
```

Comment:
[Question] Is it doesn't provide Async method or method that support cancelation?

Reply:
It does not provide async API. See reference: https://github.com/microsoft/clrmd/blob/main/src/Microsoft.Diagnostics.Runtime/DataTarget.cs